zerver/lib/users.py

```diff
@@ -476,6 +486,125 @@ def format_user_row(
     return result
 
 
+def user_access_restricted_in_realm(target_user: UserProfile) -> bool:
+    if target_user.is_bot:
+        return False
+
+    realm = target_user.realm
+    if realm.can_access_all_users_group.name == SystemGroups.EVERYONE:
+        return False
+
+    return True
+
+
+def check_user_can_access_all_users(acting_user: Optional[UserProfile]) -> bool:
+    if acting_user is None:
+        return True
```

Comment:
I guess this is the spectators case; can you add a comment on why this value is `True`?

Reply:
Yes, we [discussed](https://chat.zulip.org/#narrow/stream/101-design/topic/access.20to.20all.20users.20in.20realm.20.2310970/near/1644685) about allowing access to all the users for spectators. Would add a comment here.